### PR TITLE
fix: stop popup drag from reordering covered editor tabs (#38)

### LINKED ISSUE #38: Walkthrough item drag interaction passes through to editor tabs
When a walkthrough item is displayed over the editor file tabs, dragging the item does not move the item. Instead, the underlying editor tab receives the drag event and gets reordered.

https://github.com/user-attachments/assets/636e54de-b793-4ae5-a34f-c02c8615fdb8

Expected behavior:
Dragging a walkthrough item should move the item/popup, and the covered editor tabs should not receive the pointer event.

Actual behavior:
The file tab underneath the walkthrough item is dragged instead.

This happens when the walkthrough popup overlaps the editor tab bar.

### PR DESCRIPTION
**Pull request by forketyfork**
Closes #38

## Solution

When a walkthrough popup overlapped the editor tab bar, dragging it reordered the tab underneath instead of just moving the popup.

The editor file tabs (`JBTabsImpl`) install their own drag handler as a mouse preprocessor on the IDE glass pane. Glass-pane preprocessors receive mouse events for the whole window, independently of the normal dispatch that delivers events to the popup. So a single drag fired two handlers at once: the popup moved, and the tab handler reordered the tab beneath it.

The popup's drag and resize now run as their own glass-pane preprocessor. It registers with the default `Weighted` weight (`0`), which is lower than the tabs' handler (`2`), so the glass pane invokes it first. When a gesture starts on the popup's drag or resize handle, the preprocessor handles the move/resize itself and consumes the event, so the tab handler never sees it. Consuming is all-or-nothing per event, which is why the popup's drag had to move into the same handler that blocks the tab rather than staying on the panel listeners.

The handler is now installed once from `WalkthroughPopupSurface` when the editor is showing, which covers both the regular and diff walkthrough flows and let the diff session drop the editor-tracking plumbing it only needed for the old wiring.

Known limitation, left for a possible follow-up: a text-selection drag started inside the popup body where the body overlaps a tab can still reach the tab handler, because non-handle drags are intentionally left unconsumed so Compose text selection keeps working. In practice the drag handle covers the tab strip, so this is an edge case.

## Test plan

- [ ] Run a sandbox IDE (`just run`), open a walkthrough, and position the popup so its title/drag handle overlaps the editor tab bar. Drag it and confirm the popup moves while the tabs underneath stay in place.
- [ ] Drag the popup from the bottom-right resize handle while it overlaps the tab bar and confirm the tabs are not reordered.
- [ ] Confirm normal popup interactions still work over the tab bar: close button, navigation buttons, and the follow-up question field.
- [ ] Repeat the drag check in a diff walkthrough to confirm the diff flow behaves the same.

### CODE REVIEW

File: src/main/kotlin/com/forketyfork/walkthrough/WalkthroughPopupSurface.kt [hunk 1]
```diff
@@ -102,10 +104,24 @@ internal class WalkthroughPopupSurface(
                 pane.add(this)
             }
         }
+        installInteractionHandlerIfNeeded(editor)
```
[the remark this answers is not in the document]
Comment (forketyfork) [reply src/main/kotlin/com/forketyfork/walkthrough/WalkthroughPopupSurface.kt]: Fixed in cba29ab. `refreshInteractionHandler()` now runs on every `update()` and compares the editor's current glass pane against the one the preprocessor is registered on. When they differ (a window move, or the first time the glass pane becomes available), it disposes the old registration and installs on the new glass pane through a per-attachment child disposable, so the tabs in the new window are covered too.

File: src/main/kotlin/com/forketyfork/walkthrough/WalkthroughPopupInteraction.kt [hunk 2]
```diff
@@ -41,113 +39,132 @@ internal fun makeComponentHierarchyTransparent(component: Component?) {
...
+        when (handleRegionAt(popup, event)) {
+            PopupInteractionMode.Resize -> Cursor.getPredefinedCursor(Cursor.SE_RESIZE_CURSOR)
+            PopupInteractionMode.Drag -> Cursor.getPredefinedCursor(Cursor.MOVE_CURSOR)
+            null -> if (isWithinPopup(popup, event)) Cursor.getDefaultCursor() else null
         }
+
+    private fun reset() {
+        interactionMode = null
+        lastScreenPoint = null
     }
```
[the remark this answers is not in the document]
Comment (forketyfork) [reply src/main/kotlin/com/forketyfork/walkthrough/WalkthroughPopupInteraction.kt]: Changed in cba29ab. The glass-pane cursor is now set only for the drag and resize handles; everywhere else `resolveHandleCursor` returns null, so normal cursor resolution applies and the popup body keeps its own cursors (I-beam over selectable text, hand over links).

File: src/main/kotlin/com/forketyfork/walkthrough/WalkthroughPopupInteraction.kt [hunk 3]
```diff
@@ -41,113 +39,132 @@ internal fun makeComponentHierarchyTransparent(component: Component?) {
...
+    val content = popup.content
+    return point.x in 0..content.width && point.y in 0..content.height
+}
 
-        else -> Cursor.getDefaultCursor()
+private fun handleRegionAt(popup: WalkthroughPopupSurface, event: MouseEvent): PopupInteractionMode? {
+    val point = pointInContent(popup, event) ?: return null
+    val content = popup.content
+    val withinBounds = point.x in 0..content.width && point.y in 0..content.height
+    return when {
```
[the remark this answers is not in the document]
Comment (forketyfork) [reply src/main/kotlin/com/forketyfork/walkthrough/WalkthroughPopupInteraction.kt]: Done in cba29ab. The popup-boundary checks in `handleRegionAt` and `resizeHandleContains` now use half-open ranges (`0 until width` / `0 until height`), so a pointer one pixel past the edge counts as outside. Added tests covering the right and bottom edges.
